File: sdks/php/src/Exception/GhanaGeoException.php
<?php

declare(strict_types=1);

namespace GhanaGeo\Exception;

class GhanaGeoException extends \RuntimeException
{
    /** @param array<string, mixed> $details */
    public function __construct(
        string $message,
        public readonly string $errorCode = 'INTERNAL',
        public readonly int $statusCode = 0,
        public readonly ?string $requestId = null,
        public readonly ?string $docs = null,
        public readonly array $details = [],
        ?\Throwable $previous = null,
    ) {
        parent::__construct($message, $statusCode, $previous);
    }
}

File: sdks/dotnet/src/GhanaGeo/GhanaGeoException.cs
using System.Net;
using System.Text.Json;

namespace GhanaGeo;

public sealed class GhanaGeoException : Exception
{
    public GhanaGeoException(string code, string message, HttpStatusCode? statusCode = null,
        string? requestId = null, IReadOnlyDictionary<string, JsonElement>? details = null, string? docs = null,
        Exception? innerException = null) : base(message, innerException)
    {
        Code = code;
        StatusCode = statusCode;
        RequestId = requestId;
        Details = details;
        Docs = docs;
    }

    public string Code { get; }
    public HttpStatusCode? StatusCode { get; }
    public string? RequestId { get; }
    public IReadOnlyDictionary<string, JsonElement>? Details { get; }
    public string? Docs { get; }
}
